tests/Integration/RequestIntegrationTest.php

```diff
@@ -8,46 +8,68 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Sentry\Event;
 use Sentry\Integration\RequestIntegration;
+use Sentry\Options;
 use Zend\Diactoros\ServerRequest;
 use Zend\Diactoros\Uri;
 
 final class RequestIntegrationTest extends TestCase
 {
     /**
-     * @dataProvider invokeDataProvider
+     * @dataProvider invokeUserContextPiiDataProvider
      */
-    public function testInvoke(array $requestData, array $expectedValue): void
+    public function testInvokeWithRequestHavingIpAddress(bool $pii, array $expectedValue): void
     {
         $event = new Event();
+        $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest();
-        $request = $request->withCookieParams($requestData['cookies']);
-        $request = $request->withUri(new Uri($requestData['uri']));
-        $request = $request->withMethod($requestData['method']);
-
-        foreach ($requestData['headers'] as $name => $value) {
-            $request = $request->withHeader($name, $value);
-        }
-
+        $request = $request->withHeader('REMOTE_ADDR', '127.0.0.1');
         $this->assertInstanceOf(ServerRequestInterface::class, $request);
 
-        RequestIntegration::applyToEvent($event, $request);
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $pii]));
+        RequestIntegration::applyToEvent($integration, $event, $request);
 
-        $this->assertEquals($expectedValue, $event->getRequest());
+        $this->assertEquals($expectedValue, $event->getUserContext()->toArray());
     }
 
-    public function testInvokeWithRequestHavingIpAddress(): void
+    public function invokeUserContextPiiDataProvider(): array
     {
-        $event = new Event();
-        $event->getUserContext()->setData(['foo' => 'bar']);
+        return [
+            [
+                true,
+                ['ip_address' => '127.0.0.1', 'foo' => 'bar'],
+            ],
+            [
+                false,
+                ['foo' => 'bar'],
+            ],
+        ];
+    }
 
-        $request = new ServerRequest();
-        $request = $request->withHeader('REMOTE_ADDR', '127.0.0.1');
-        $this->assertInstanceOf(ServerRequestInterface::class, $request);
+    /**
+     * @dataProvider invokeDataProvider
+     */
+    public function testInvoke(array $requestData, array $expectedValueWithPii, array $expectedValueWithoutPii): void
+    {
+        foreach ([true, false] as $pii) {
```

Comment:
Instead of using a `foreach` you should use the data provider as you do with the rest of the data

Reply:
I didn't want to duplicate the data in the `dataProvider` and just toggle the flag true / false.
Maybe if I box the data differently this will work.

Reply:
The data provider should provide the data for a complete test case, if half of the data is coming from the provider and half is inside the test it's confusing and it could means that either the method is testing too much things together or the provided data is not enough for the test-case